### PR TITLE
Carl ansi part1

### PR DESCRIPTION
**Pull request by ccodere**
Summary of changes in this pull request:
Pascal compiler:
* ANSI C conversion
* Cleanup to reduce number of warnings.
* fix some wrong tab/space issues
 
ANSI C compiler:
* Bugfix of bug on fprint in error reporting in C compiler from last pull request (wrong file handle)
* Some small cleanup to reduce number of warnings.

What is the policy for the C code in general? BSD Allman is ok with tabs or spaces? Seems existing code uses tabs, right?

### CODE REVIEW

**Comment by davidgiven** [comment]
I haven't had a chance to look through this yet, but re formatting there's a clang-format file in the root.

**Comment by ccodere** [comment]
> I haven't had a chance to look through this yet, but re formatting there's a clang-format file in the root.

Thanks for the information, it gave me the time to fix 2 issues in my pull request.

File: lang/pc/comp/LLlex.c [hunk 1]
```diff
@@ -153,9 +153,7 @@ SkipComment()
 	}
 }
 
-STATIC struct string *
-GetString( delim )
-register int delim;
+static struct string *GetString(register int delim)
```
Comment (davidgiven) [inline lang/pc/comp/LLlex.c:156]: This is not a request for a change, just a comment: no compiler, not even the ACK, honours `register` any more. Might as well delete 'em if they're in the way.

**Comment by davidgiven** [comment]
Looks fine to me --- thank you very much! The entire build log now fits in the Travis console, at least for Linux... and that hasn't happened for a long time!

**Pull request merged**